spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -419,5 +420,8 @@ public void testFaultToleranceOnWrite() throws IOException {
 
     assertThat(snapshotBeforeFailingWrite).isEqualTo(snapshotAfterFailingWrite);
     assertThat(resultBeforeFailingWrite).isEqualTo(resultAfterFailingWrite);
+
+    FileUtils.deleteDirectory(location);
```

Comment:
thank you @manuzhang , can you expand a bit on how this fix the flaky tests?

Reply:
Shouldn't we just be using a temp directory fixture from Asserts?

Reply:
>     java.io.IOException: Failed to delete temp directory /tmp/junit8883793498240284997. The following paths could not be deleted (see suppressed exceptions for details): parquet/test/data/00001-127-9abadc06-9aa2-4b3b-8e77-66af8ddb7a3e-0-00001.parquet

The flakiness is because table data cannot be deleted sometimes. This PR ensures deleting table data before temp directory cleaning up itself.